libfwupdplugin/fu-common.c

```diff
@@ -2401,15 +2405,20 @@ fu_common_get_esp_default (GError **error)
 	g_autoptr(GPtrArray) volumes_fstab = g_ptr_array_new ();
 	g_autoptr(GPtrArray) volumes_mtab = g_ptr_array_new ();
 	g_autoptr(GPtrArray) volumes = NULL;
+	g_autoptr(GError) error_local = NULL;
 
 	/* for the test suite use local directory for ESP */
 	path_tmp = g_getenv ("FWUPD_UEFI_ESP_PATH");
 	if (path_tmp != NULL)
 		return fu_volume_new_from_mount_path (path_tmp);
 
-	volumes = fu_common_get_volumes_by_kind (FU_VOLUME_KIND_ESP, error);
-	if (volumes == NULL)
-		return NULL;
+	volumes = fu_common_get_volumes_by_kind (FU_VOLUME_KIND_ESP, &error_local);
+	if (volumes == NULL) {
+		g_debug ("%s, falling back to %s", error_local->message, FU_VOLUME_KIND_FAT32);
+		volumes = fu_common_get_volumes_by_kind (FU_VOLUME_KIND_FAT32, error);
+		if (volumes == NULL)
+			return NULL;
```

Comment:
I do wonder if we should combine the errors before returning? e.g. `g_prefix_error (error, "%s: ", error_local->message);` -- otherwise you're going to get a message about not finding the fallback and not actually have a clue we really wanted the ESP kind.

Reply:
fixed